spec/lib/figaro_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe Figaro::Environment do
+  let(:config) do
+    {
+      'test_key' => 'value',
+      'production' => {
+        'test_key' => 'overridden value',
+      },
+    }
+  end
+
+  let(:config_environment) { 'test' }
+
+  describe '#initialize' do
+    it 'warns and uses ENV when key is set in ENV and file config' do
+      allow(ENV).to receive(:[]).with('test_key').and_return('overridden value')
```

Comment:
I think we're better off doing something like:

```
let(:env) { { 'test_key' => 'aaa' } }
before do
  stub_const('ENV', env)
end

# ...
```

and I'm specifically looking for a spec that checks for side effects by writing to ENV in `#setup`

```ruby
expect(ENV['SOME_KEY']).to be_nil
setup
expect(ENV['SOME_KEY').to eq('value')
```